G1ANT.Addon.Azure/Extensions/TaskExtensions.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace G1ANT.Addon.Azure.Extensions
+{
+    public static class TaskExtensions
+    {
+        private struct VoidTypeStruct { } //dummy struct - there is no non-generic version of TaskCompletionSource<TResult>
+        public static Task TimeoutAfter(this Task task, int millisecondsTimeout)
+        {
+            if (task.IsCompleted || millisecondsTimeout == Timeout.Infinite)
+            {
+                return task;
+            }
+
+            var completionSource = new TaskCompletionSource<VoidTypeStruct>();
+
+            if (millisecondsTimeout == 0)
+            {
+                completionSource.SetException(new TimeoutException());
+                return completionSource.Task;
+            }
+
+            var timeoutTimer = new Timer(state => SetTimerTimeoutException(state), completionSource, millisecondsTimeout, Timeout.Infinite);
+
+            task.ContinueWith((antecedent, state) => 
+                FinishTaskExecution(antecedent, state),
+                Tuple.Create(timeoutTimer, completionSource),
+                CancellationToken.None,
+                TaskContinuationOptions.ExecuteSynchronously,
+                TaskScheduler.Default);
+
+            return completionSource.Task;
+        }
+
+        internal static void MarshalTaskResults<TResult>(
```

Comment:
No need for this line break, especially that line below is incorrectly indented.
Why is this method internal and not private/public?